### PR TITLE
destination-redshift: add option for drop cascade

### PR DESCRIPTION
**Pull request by stephane-airbyte**
fixes https://github.com/airbytehq/oncall/issues/5226

### CODE REVIEW

**Comment by stephane-airbyte** [comment]
* **#38189** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/38189?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @stephane-airbyte and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by edgao — APPROVED** [review]
nit: pr title should be redshift, and I think you linked the wrong issue in the description?

otherwise, one comment about structure, lgtm assuming CI succeeds

**Review by evantahler — COMMENTED** [review]
This is great - it gives users the option to self-solve... but:
1. we still need to catch the error thrown in this case and turn it into a config_error, not a system_error
2. I would have expected to see the spec change to include the new option?

File: airbyte-integrations/connectors/destination-redshift/src/main/java/io/airbyte/integrations/destination/redshift/RedshiftStagingS3Destination.java [hunk 1]
```diff
@@ -179,7 +180,10 @@ public JsonNode toJdbcConfig(final JsonNode config) {
 
   @Override
   protected JdbcSqlGenerator getSqlGenerator(final JsonNode config) {
-    return new RedshiftSqlGenerator(getNamingResolver());
+
+    final JsonNode dropCascadeNode = config.get(RedshiftDestinationConstants.DROP_CASCADE_OPTION);
```
[the remark this answers is not in the document]
Comment (stephane-airbyte) [reply airbyte-integrations/connectors/destination-redshift/src/main/java/io/airbyte/integrations/destination/redshift/RedshiftStagingS3Destination.java]: done

**Comment by stephane-airbyte** [comment]
@evan: indeed. My PRs got mixed up on my local...

**Review by edgao — APPROVED** [review]

**Review by evantahler — APPROVED** [review]

**Comment by stephane-airbyte** [comment]
### Merge activity

* **May 14, 2:54 PM EDT**: @stephane-airbyte merged this pull request with [Graphite](https://app.graphite.dev/github/pr/airbytehq/airbyte/38189).

**Pull request merged**